### PR TITLE
Tasks/WP-655 Add support for User Projects in apps

### PR DESCRIPTION
**Pull request by chandra-tacc**
## Overview: ##
For Express apps, user would like to have all their projects mounted and available.
Follow the solution from v2, where if _UserProject env var is available, then the list of projects is appended in this format:
uuid1, projectid2 uuid1, projectid2
upto 500.

QGIS app is updated to handle mapping the values.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [WP-655](https://tacc-main.atlassian.net/browse/WP-655)

## Testing Steps: ##
1. Go to https://designsafe.dev/rw/workspace/qgis_express?appVersion=3.36. Submit job and see if your projects are in MyProjects folder.
2. Go to https://designsafe.dev/rw/workspace/openfoam to test nothing is broken regression wise and job submission works. This app has env vars and making sure nothing in existing env var breaks

## UI Photos:
Screenshot 2024-08-14 at 2 39 46 PM

Screenshot 2024-08-14 at 2 53 04 PM

## Notes: ##
Followups:
* I move the script to create project mounts into a shared one - so it can be reused

### CODE REVIEW

File: designsafe/apps/workspace/api/views.py [hunk 1]
```diff
@@ -777,6 +777,23 @@ def _submit_job(self, request, body, tapis, username):
...
+                projects = request.user.projects.order_by("-last_updated")
+                entry["value"] = " ".join(
+                    f"{project.uuid},{project.value['projectId'] if project.value['projectId'] != 'None' else project.uuid}"
+                    for project in projects[:settings.USER_PROJECTS_LIMIT]
+                )
+                projects_updated = True
+                break
+
+        if projects_updated:
+            job_post["parameterSet"]["envVariables"] = env_variables
```
Comment (rstijerina) [inline designsafe/apps/workspace/api/views.py:795]: Could this go right after line 790, and no need for `projects_updated` var?

**Review by rstijerina — APPROVED** [review]
LGTM, just one minor comment but not a blocker

File: designsafe/apps/workspace/api/views.py (hunk 1 above)
Comment (chandra-tacc) [reply designsafe/apps/workspace/api/views.py]: Yes, addressed this and tested again.

**Review by fnets — APPROVED** [review]
LGTM, but I wasn't able to test this locally

**Pull request merged**